bundler/lib/bundler/fetcher.rb

```diff
@@ -5,6 +5,7 @@
 require "securerandom"
 require "zlib"
 require "rubygems/request"
+require "rubygems/ci_detector"
```

Comment:
It is possible to run old RubyGems with new Bundler and this file could be missing. What to do in that case? Should we keep the current logic around as a fallback for some time? @deivid-rodriguez any idea what's common practice?

Reply:
That.. actually probably explains the problem I'm trying to resolve locally and in CI right now >.<

Reply:
Does make any shared-code plan kind of awkward though. I'm sure it's further than *I* ought to go, but it seems like there might be space for a shared library of some sort, which would get packaged with both systems?

Reply:
I guess keeping the existing bundler/fetcher code for a while makes sense - should I check `Gem.autoload?` for that?

Reply:
This is where the solutions to this type of problem live:

https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/rubygems_ext.rb

We have to be careful not to make a huge mess though. I'm not sure if duplicating the code into here is better or worse.

Reply:
This doesn't look.. _central_ enough to belong in there, I think. What's the cross-version compatibility policy? If it's a question of keeping an extra copy of one of these methods for a few releases, that seems fine to me, but if we're talking about a long term situation then I'm not sure this ticket is worth resolving :-/

Though I think the research/documentation and the data fixes are worth keeping anyway, so I'd prefer to just split it into parts and keep the parts with the projects that use them. And possibly extract the big comment into a markdown file they both reference?

The other idea that occurs to me is to drive the behaviour from a config file, and include that in both builds. But the logic for memoizing the parsed config would be larger than what we're replacing, so it would seem silly.

Reply:
A similar thought occurred to me - my idea was just to _enforce_ that they match in CI, rather than to generate one from the other, since I feel like machine-generated commits make for complicated workflows. Would that appeal to you? It feels less complex than an actual generator, and the file's not really large enough that it'll hurt much to have to duplicate changes across on the occasion that they're made..

Reply:
I like your idea of writing a test that makes sure we keep them synced. If someone is editing one, the test will help make sure they fix the other one.

Reply:
Cool, I'll get a version of that up in a bit :-)

Reply:
Alright, took a couple cycles to understand how the path helpers worked (`ruby_core?`), but that should do it - there are now `Gem::CIDetector` and `Bundler::CIDetector`; the spec for the later doesn't test functionality, just textual equivalence.